contracts/contracts/strategies/balancer/BaseBalancerStrategy.sol

```diff
@@ -225,28 +225,37 @@ abstract contract BaseBalancerStrategy is InitializableAbstractStrategy {
 
     /* solhint-disable max-line-length */
     /**
-     * @notice BPT price is calculated by dividing the pool (sometimes wrapped) market price by the
-     * rateProviderRate of that asset. To get BPT expected we need to multiply that by underlying
-     * asset amount divided by BPT token rate. BPT token rate is similar to Curve's virtual_price
-     * and expresses how much has the price of BPT appreciated in relation to the underlying assets.
+     * @notice BPT price is calculated by taking the rate from the rateProvider of the asset in
+     * question. If one does not exist it defaults to 1e18. To get the final BPT expected that
+     * is multiplied by the underlying asset amount divided by BPT token rate. BPT token rate is
+     * similar to Curve's virtual_price and expresses how much has the price of BPT appreciated
+     * (e.g. due to swap fees) in relation to the underlying assets
      *
-     * @dev
-     * bptPrice = pool_asset_oracle_price / pool_asset_rate
+     * Using the above approach makes the strategy vulnerable to a possible MEV attack using
+     * flash loan to manipulate the pool before a deposit/withdrawal since the function ignores
+     * market values of the assets being priced in BPT.
+     *
+     * At the time of writing there is no safe on-chain approach to pricing BPT in a way that it
+     * would make it invulnerable to MEV pool manipulation. See recent Balancer exploit:
+     * https://www.notion.so/originprotocol/Balancer-OETH-strategy-9becdea132704e588782a919d7d471eb?pvs=4#1cf07de12fc64f1888072321e0644348
      *
-     * Since we only have oracle prices for the unwrapped version of the assets the equation
-     * turns into:
+     * To mitigate MEV possibilities VaultValueChecker in conjunction with checkBalance will catch
```

Comment:
This sentence could be "To mitigate MEV possibilities during deposits and withdraws, the VaultValueChecker will use checkBalance before and after the move to ensure the expected changes took place."

Reply:
thanks much better!